tests/queries/0_stateless/02710_show_table.sql

```diff
@@ -0,0 +1,15 @@
+DROP TABLE IF EXISTS t_2710_show_table;
+
+CREATE TABLE t_2710_show_table(n1 UInt32, s String) engine=Log;
```

Comment:
I tried your PR and checked these variants:

* Tables: (after CREATE TABLE t ...)
  1. show create table t
  2. show create t
  3. show table t

* Databases:  (after CREATE DATABASE d ...)
  1. show create database d
  2. show create d (*)
  3. show database d

* Views: (after CREATE VIEW v ...)
  1. show create view v
  2. show create v
  3. show view v

* Dictionaries: (after CREATE DICTIONARY d ...)
  1. show create dictionary d
  2. show create d
  3. show dictionary d

All variants work except (*) - that's kind of unexpected. I understand that if the object type is not specified, some search order (e.g. 1. tables 2. databases 3. views 4. dictionaries) is needed. Maybe this special case could be implemented as well (for consistency)?

Reply:
Could implement, but will make the code ugly somehow, `SHOW CREATE` syntax support VIEW and DICTIONARY because they are a kind of special tables.